File: Greedy_Algorithm/C++/백준_그리디알고리즘_ATM(11399).cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    int people;
    int minute;
    int sum = 0;
    int result = 0;
    vector<int> vec;
    cin >> people;
    for (int i = 0; i < people; i++) {  
        cin >> minute;
        vec.push_back(minute);
    }
    sort(vec.begin(), vec.end());
    for (int i = 0; i < people; i++) {
        sum += vec[i];
        result += sum;
    }
    cout << result << endl;
    return 0;
}

File: Greedy_Algorithm/Python/백준_그리디알고리즘_ATM(11399).py
# baekjoon 11399 ATM
# ATM 1대
# 사람은 1번 부터 N번 까지 번호가 매겨져있다.
# i번 사람이 돈을 인출하는데 걸리는 시간은 Pi분이다.
# 작업 순서가 작은대로 정렬한다.
# 그리고 누적합을 계산한다.
# 돈을 인출하는데 필요한 시간의 합을 '최소'로 만든다.
people = int(input())
minute = list(map(int,input().split()))
minute.sort()
sum=0
result=0
for i in range(people):
    sum += minute[i]
    result += sum
print(result)
